test/parallel/test-env-newprotomethod-remove-unnecessary-prototypes.js

```diff
@@ -0,0 +1,15 @@
+'use strict';
+require('../common');
+require('../common/fixtures');
+
+// This test ensures that unnecessary prototypes are no longer
+// being genrated by Environment::NewFunctionTemplate
+
+const assert = require('assert');
+
+
+assert.strictEqual((typeof process
+.binding('udp_wrap').UDP.prototype.bind6.prototype),
+                   'undefined',
+                   'Unnecessary prototype "process.binding("udp_wrap").\
+UDP.prototype.bind6.prototype" was generated!');
```

Comment:
@bnoordhuis I have verified that my tests pass `make lint`. I also cut down the messages in my asserts to say only the names of the prototypes that they are testing, and nothing else, to avoid bloat. Furthermore, I added a few more method tests, including tests for `process.binding('tcp_wrap').TCP`, and some `protoype in` tests as well.

Please let me know if there are any further improvements needed.